main.tf

```diff
@@ -120,13 +120,38 @@ resource "aws_db_instance" "db_including_name" {
   storage_encrypted           = "${var.storage_encrypted}"
   storage_type                = "${var.storage_type}"
   username                    = "${var.database_user}"
+  tags                        = "${merge(var.tags, map("Name", format("%s-%s", var.environment, var.name)), map("Env", var.environment))}"
+}
 
-  tags = "${merge(var.tags, map("Name", format("%s-%s", var.environment, var.name)), map("Env", var.environment))}"
+# The database instance itself with read replica
+resource "aws_db_instance" "db_read_replica" {
+  count = "${var.database_name == "" && var.engine_type != "aurora" && var.engine_type != "aurora-mysql" && var.engine_type != "aurora-postgresql" && var.replicate_source_db != "" ? 1 : 0}"
```

Comment:
This can be a lot simpler. You can just check whether or not `replicate_source_db` is empty or not. i.e. `count = "${var.replicate_source_db != "" ? 1 : 0}"`